### PR TITLE
UCP: wrap function to open interface on specific device

### PR DESCRIPTION
**Pull request by changchengx**
## What
[UCP/CORE: Removed unused sockaddr code](https://github.com/openucx/ucx/pull/8752) removed the sockaddr open iface mode, then it only support ```UCT_IFACE_OPEN_MODE_DEVICE``` open iface mode. This PR wrapper the iface open parameters into ```ucp_worker_iface_open```.

### CODE REVIEW

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -1308,53 +1298,59 @@ ucs_status_t ucp_worker_iface_open(ucp_worker_h worker, ucp_rsc_index_t tl_id,
...
+        iface_params.field_mask     |= UCT_IFACE_PARAM_FIELD_AM_ALIGNMENT |
+                                       UCT_IFACE_PARAM_FIELD_AM_ALIGN_OFFSET;
+        iface_params.am_align_offset = sizeof(ucp_am_hdr_t);
+        iface_params.am_alignment    = worker->am.alignment;
     }
 
-    iface_params->field_mask |= UCT_IFACE_PARAM_FIELD_FEATURES;
-    iface_params->features    = ucp_worker_get_uct_features(worker->context);
+    iface_params.field_mask |= UCT_IFACE_PARAM_FIELD_FEATURES;
+    iface_params.features    = ucp_worker_get_uct_features(worker->context);
```
Comment (rakhmets) [inline src/ucp/core/ucp_worker.c:1350]:
```suggestion
    iface_params.features    = ucp_worker_get_uct_features(context);
```

**Review by rakhmets — APPROVED** [review]

**Review by rakhmets — DISMISSED** [review]

**Comment by Alexey-Rivkin** [comment]
bot:retest

File: src/uct/base/uct_md.c [hunk 2]
```diff
@@ -268,9 +268,6 @@ ucs_status_t uct_iface_open(uct_md_h md, uct_worker_h worker,
 
     if (params->open_mode & UCT_IFACE_OPEN_MODE_DEVICE) {
         tl = uct_find_tl(md->component, params->mode.device.tl_name);
-    } else if ((params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_CLIENT) ||
-               (params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER)) {
-        tl = uct_find_tl(md->component, NULL);
```
Comment (yosefe) [inline src/uct/base/uct_md.c:273]: this is another change - you are deprecating UCT_IFACE_OPEN_MODE_SOCKADDR_CLIENT and UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER flags in UCT API
better move it to another PR

Comment (changchengx) [reply src/uct/base/uct_md.c]: I've reverted this change. Waiting for CI result.

**Review by yosefe — APPROVED** [review]
pls squash

**Review by yosefe — APPROVED** [review]
